### DISCUSSION TITLE
Unable to set Authorization header

### DISCUSSION THREAD

**Discussion by quinnjr** [enisdenjo/graphql-ws#588] [Q&A] [answered]
Hello,

I'm trying connect my graphql client to a subscription that has a JWT auth guard on it. Currently, I'm noticing that nothing I'm trying to do in regards to sending the Authorization header is working. Chrome inspector on the network request shows that the Authorization header is not being set.

My configuration of the client:

```
const wsLink = new GraphQLWsLink(
  createClient({
    url: environment.wsUrl + '/graphql',
    connectionParams: () => {
        const token = storageService.getItem('accessToken');
        console.log(token);
        if (!token) {
          return {};
       }
      return {
        headers: {
          authorization: token ? `Bearer ${token}` : '',
          Authorization: token ? `Bearer ${token}` : ''
        }
      };
    },
    retryAttempts: 5,
    shouldRetry: (error) => {
      if (!environment.production) {
        console.log('Websocket error:', error);
     }
     return true;
   },
   keepAlive: 10000
   })
);
```

`token` is properly being set.

**Comment by enisdenjo** [comment, marked as answer]
Hey, you cannot manipulate the [HTTP Upgrade request headers in the browser](https://stackoverflow.com/questions/4361173/http-headers-in-websockets-client-api/4361358#4361358). The `connectionParams` are passed through the `ConnectionInit` message's payload inside an established WebSocket connection.
